src/ape/api/providers.py

```diff
@@ -133,6 +136,33 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockGasAPI:
+    @property
+    def base_fee(self) -> int:
```

Comment:
Now, it will always exist but just raise `NotImpementedError` by default, just like the `ProviderAPI`.

This way, the real implementation does not have to have a return type of `Optional`, so that is a win.

Reply:
This works too

Reply:
99.9% of the time, people will only be calling this on the latest block while crafting a dynamic transaction